src/voodoo/spellbook/spells/stoneskin.ts

```diff
@@ -50,8 +50,9 @@ export const stoneskin: SpellFunction = async (voodoo, accountId, upgradeConfigs
 
   for (const playerId of playerIds) {
     const baseDamageProtection = await voodoo.getPlayerCheckStatBase({ accountId: playerId, stat: 'damageprotection' });
+    const currentDamageProtection = await voodoo.getPlayerCheckStatCurrent({ accountId: playerId, stat: 'damageprotection' })
 
-    if (baseDamageProtection) {
+    if (baseDamageProtection == currentDamageProtection) {
       const buffedDamageProtection = baseDamageProtection * multiplier;
```

Comment:
Implementing mdigena's suggestions
```suggestion
      const damageprotectionBuff = baseDamageProtection * bonus;
```